cmd/gocharm/main.go

```diff
@@ -184,6 +188,9 @@ func main1(pkgPath string) error {
 		return errgo.Mask(err)
 	}
 	for name := range allowed {
+		if name == "src" && *noSource && !*source {
```

Comment:
Given that if *noSource is true, *source must be false, I think this is equivalent to:

    if *noSource && name == "src" {
        continue
    }

(I also swapped the two exprs as I think it reads more nicely that way)